pallets/offchain-duster/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+// This file is part of HydraDX.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![allow(clippy::upper_case_acronyms)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use orml_traits::GetByKey;
+
+use frame_system::offchain::{CreateSignedTransaction, SendSignedTransaction, Signer};
+
+use frame_support::sp_runtime::offchain::storage_lock::{StorageLock, Time};
+use frame_support::sp_runtime::offchain::Duration;
+use orml_utilities::OffchainErr;
+use sp_core::crypto::KeyTypeId;
+use sp_runtime::traits::Saturating;
+
+pub use pallet_duster::Call as DusterCall;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+pub const OFFCHAIN_WORKER_LOCK: &[u8] = b"hydradx/offchain-duster/lock/";
+pub const LOCK_DURATION: u64 = 100;
```

Comment:
Is this ms? If yes isn't it too short? We should probably do this only every x blocks

Reply:
yes. but it is for one dust_account call. Lock guard is extended after each call. 

It is because it is hard to determine correct duration as it may vary.